src/bridge/test-packages.c

```diff
@@ -773,6 +778,66 @@ test_glob (TestCase *tc,
   cockpit_assert_bytes_eq (message, "b\n", 2);
 }
 
+static const Fixture fixture_with_gzip = {
+    .datadirs = { SRCDIR "/src/bridge/mock-resource/gzip", NULL },
+    .path = "/package/file.txt",
+    .binary = TRUE,
+    .headers = { "Accept-Encoding", "*" },
+};
+
+static void
+test_with_gzip (TestCase *tc,
+                gconstpointer fixture)
+{
+  GError *error = NULL;
+  GBytes *message;
+  JsonObject *object;
+  GBytes *data;
+
+  while (tc->closed == FALSE)
+    g_main_context_iteration (NULL, TRUE);
+  g_assert_cmpstr (tc->problem, ==, NULL);
+
+  message = mock_transport_pop_channel (tc->transport, "444");
+  object = cockpit_json_parse_bytes (message, &error);
+  g_assert_no_error (error);
+  cockpit_assert_json_eq (object, "{\"status\":200,\"reason\":\"OK\",\"headers\":{" STATIC_HEADERS ",\"X-Cockpit-Pkg-Checksum\":\"" CHECKSUM_GZIP "\",\"Content-Encoding\":\"gzip\",\"Content-Type\":\"text/plain\"}}");
+  json_object_unref (object);
+
+  data = mock_transport_combine_output (tc->transport, "444", NULL);
+  g_assert_cmpint (g_bytes_get_size (data), ==, 9377);
```

Comment:
Doesn't this leak `data`? That may be the reason for the semaphore failure? Note that we make plenty of use of `g_autoptr()` now, which lets the compiler deal with these boring issues. :-)

Reply:
Thanks. Fixed above.